doc generation broken on master

At ea5af50bd14e6d132fd86dd0ceebc7de4f715352, in our packaging, I see:
```
...
| -- validating generated_list
| -- validating generator
| -- validating inc
Traceback (most recent call last):
  File "/var/tmp/portage/dev-build/meson-9999/work/meson-9999/docs/./genrefman.py", line 15, in <module>
    raise SystemExit(main())
                     ~~~~^^
  File "/var/tmp/portage/dev-build/meson-9999/work/meson-9999/docs/refman/main.py", line 51, in main
    refMan = loader.load()
  File "/var/tmp/portage/dev-build/meson-9999/work/meson-9999/docs/refman/loaderbase.py", line 205, in load
    return resolver.validate_and_resolve(manual)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/var/tmp/portage/dev-build/meson-9999/work/meson-9999/docs/refman/loaderbase.py", line 171, in validate_and_resolve
    self._validate_func(m)
    ~~~~~~~~~~~~~~~~~~~^^^
  File "/var/tmp/portage/dev-build/meson-9999/work/meson-9999/docs/refman/loaderbase.py", line 85, in _validate_func
    func.returns = self._resolve_type(func.returns.raw)
                   ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/var/tmp/portage/dev-build/meson-9999/work/meson-9999/docs/refman/loaderbase.py", line 71, in _resolve_type
    assert base_type in self.type_map, f'No known object {t}'
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: No known object list[str]
ninja: build stopped: subcommand failed.
 * ERROR: dev-build/meson-9999::gentoo failed (compile phase):
 *   ninja -v -C docs/builddir failed
```

The last commit I built was dea8d2d1116416d0105657321da4e0eb25794e42.

--- a/docs/validatelinks.py
+++ b/docs/validatelinks.py
@@ -28,14 +28,20 @@ async def main(filename):
             tasks = []
             for link in LINK.finditer(text):
                 name, url = link.groups()
+                name = name.replace('\n', ' ')
                 task = asyncio.ensure_future(fetch(session, name, url, timeout))
                 tasks.append(task)
             responses = asyncio.gather(*tasks)
             errors = [r for r in await responses if r is not None]
+        bad = False
         for name, url, result in errors:
-            print(f'"{name}" {url} {result}')
+            if re.match(r'https://github.com/(search|topics/)', url) and result == 429:
+                print(f'"{name}" {url} {result} (ignored)')
+            else:
+                print(f'"{name}" {url} {result}')
+                bad = True
         if errors:
-            sys.exit(1)
+            sys.exit(int(bad))
 
 
 if __name__ == '__main__':

--- a/docs/yaml/objects/inc.yaml
+++ b/docs/yaml/objects/inc.yaml
@@ -4,7 +4,7 @@ description: Opaque wrapper for storing include directories
 
 methods:
 - name: to_list
-  returns: list[str]
+  returns: array[str]
   since: 1.12.0
   description: |
     Returns a list of strings of absolute paths to the directories

